### PR TITLE
Adding changelog and CI check for changes on version bump

### PR DESCRIPTION
**Pull request by sydney-runkle**
Fix https://github.com/pydantic/logfire/issues/237

### CODE REVIEW

**Comment by Kludex** [comment]
Can we add the changelog to the docs, like this: https://github.com/encode/uvicorn/blob/master/docs/release-notes.md ?

Or just have the changelog on the docs itself?

**Comment by sydney-runkle** [comment]
Please review

File: .github/workflows/check_changelog.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        run: |
+          python -m pip install toml
+          for file in pyproject.toml HISTORY.md; do
+              base_name=$(basename $file .${file##*.})
+              if git diff --name-only HEAD^ | grep -q "^$file$"; then
+                echo "${base_name}_changed=true" >> $GITHUB_ENV
+              else
+                echo "${base_name}_changed=false" >> $GITHUB_ENV
+              fi
+          done
```
Comment (Kludex) [inline .github/workflows/check_changelog.yml]: You don't need to install toml on python 3.12 to read TOML files, you can use https://docs.python.org/3/library/tomllib.html, which is available from 3.11.

**Review by Kludex — CHANGES_REQUESTED** [review]

**Comment by Kludex** [comment]
Feel free to merge it after removing the toml dependency 👍

**Pull request merged**